tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,216 @@
+
+  auto result_dc = dc.init(ComponentConfiguration(digital_conf));
+  ASSERT_EQ(Result::RESULT_OK, result_dc);
+
+  result_dc = dc.start();
+  ASSERT_EQ(Result::RESULT_OK, result_dc);
+
+  result_dc = dc.connect(ConnectionConfiguration());
+  ASSERT_EQ(Result::RESULT_OK, result_dc);
+
+  result_dc = dc.check();
+  ASSERT_EQ(Result::RESULT_OK, result_dc);
+
+  // Access the dynamic properties from the SignalGeneratorComponent
+  DynamicPropertyAccess dynamic_property_access_signal("/signal"_uri);
+  DynamicPropertyAccess dynamic_property_access_digital("/digital"_uri);
+  
+  float sm_not_exits, dc_not_exits;;
+  ASSERT_ANY_THROW(dynamic_property_access_digital.get("not_exists", sm_not_exits));
+  ASSERT_ANY_THROW(dynamic_property_access_digital.get("not_exists", dc_not_exits));
+ 
+  double signal_amplitude, digital_amplitude, frequency, threshold;
+  bool cosine;
+
+  ASSERT_TRUE(dynamic_property_access_signal.get("amplitude", signal_amplitude));
+  ASSERT_TRUE(dynamic_property_access_signal.get("frequency", frequency));
+  ASSERT_TRUE(dynamic_property_access_signal.get("cosine", cosine));
+  ASSERT_TRUE(dynamic_property_access_digital.get("threshold", threshold));
+  ASSERT_TRUE(dynamic_property_access_digital.get("amplitude", digital_amplitude));
+
+  ASSERT_EQ(0, signal_amplitude);
+  ASSERT_EQ(0, frequency);
+  ASSERT_EQ(0, cosine);
+  ASSERT_EQ(0, threshold);
+  ASSERT_EQ(0, digital_amplitude);
+}
+
+//-------------------------------------------------
+// Create test for SignalGeneratorComponent
+//-------------------------------------------------
+TEST (ExampleTest, SignalGeneratorComponentCheck)
+{
+  //Configuration
+  auto signal_conf = "{"
+      "outputs: {"
+        "value: { protocol: 'api' },"
+        "timestamp: { protocol: 'api' },"
+        "frame_id: { protocol: 'api' },"
+        "fix_dynamic: { protocol: 'api' }"
+      "}"
+    "}";
+
+  //SignalGeneratorComponent state machine
+  StateMachine sm(std::make_shared<SignalGeneratorComponent>("/signal"_uri));
+ 
+  auto result = sm.init(ComponentConfiguration(signal_conf));
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.start();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.connect(ConnectionConfiguration());
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.check();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  //Fake input channels
+  auto inputSignal = std::make_shared<FakeInput<Signal>>("/signal"_uri, "api", "/signal/value");
+  ASSERT_TRUE(inputSignal->connect());
+
+  //TEST
+  sm.update();
+
+  auto valueSignal = std::shared_ptr<const Signal>();
+
+  ReceiveStatus rec_signal = inputSignal->receive(valueSignal, 500ms);
+
+  ASSERT_EQ(ReceiveStatus::RECEIVE_OK, rec_signal);
+  EXPECT_EQ(valueSignal->value, 0);
+
+  result = sm.reset();
+}
+
+//-------------------------------------------------
+// Create test for DigitalConverterComponent
+//-------------------------------------------------
+TEST (ExampleTest, DigitalConverterComponentCheck)
```

Comment:
Here you need to check if the output of `DigitalConverterComponent` is corrected. Due to use a spinner to asynchronous write the digital converter output, it is difficult to check the exactly value. However, you can check if the output is between a specific range. For example, greater than 0 and less than 1.